scipy/stats/tests/test_distributions.py

```diff
@@ -3089,6 +3089,46 @@ def test_moments_stats(self, r):
         # r-th non-central moment is non-finite (inf or nan) if r >= c.
         assert not np.any(np.isfinite(stats.loglaplace.stats(c, moments=mom)))
 
+    @pytest.mark.parametrize("c", [0.5, 1.0, 2.0])
+    @pytest.mark.parametrize("loc, scale", [(-1.2, 3.45)])
+    @pytest.mark.parametrize("fix_c", [True, False])
+    @pytest.mark.parametrize("fix_loc", [True])
+    @pytest.mark.parametrize("fix_scale", [True, False])
+    def test_fit_analytic_mle(self, c, loc, scale, fix_c, fix_loc, fix_scale):
+        # Test that the analytic MLE produces the same result as the generic
+        # (numerical) MLE.
+
+        rng = np.random.default_rng(6762668991392531563)
+        data = stats.loglaplace.rvs(c, loc=loc, scale=scale, size=100,
+                                    random_state=rng)
+
+        kwds = {}
+        if fix_c:
+            kwds['fc'] = c
+        if fix_loc:
+            kwds['floc'] = loc
+        if fix_scale:
+            kwds['fscale'] = scale
+        nfree = 3 - len(kwds)
+
+        if nfree == 0:
+            error_msg = "All parameters fixed. There is nothing to optimize."
+            with pytest.raises((RuntimeError, ValueError), match=error_msg):
+                stats.loglaplace.fit(data, method='mle', **kwds)
+            return
+
+        generic_args = super(type(stats.loglaplace),
+                             stats.loglaplace).fit(data, **kwds, method='mle')
+        analytic_args = stats.loglaplace.fit(data, **kwds, method='mle')
+
+        # Sanity check that the specialized MLE is actually executed.
+        # Due to floating point arithmetic, the generic MLE is unlikely
+        # to produce the exact same result as the analytic MLE.
+        assert np.any(generic_args != analytic_args)
```

Comment:
There is a function [`assert_less_or_close_loglike`](https://github.com/scipy/scipy/blob/4edfcaa3ce8a387450b6efce968572def71be089/scipy/stats/tests/test_distributions.py#L251) that takes care of the comparison. Could you add this check that they are not exactly equal to it (and use it)?

Reply:
Sure let me do that.

Reply:
Got 44 test failures after that change… I’ll look into whether they’re expected (some methods fall back to super mle when it finds difficulty) or the sanity check is too speculative.

Reply:
Hmm that would make sense. Could make that check opt-in.

Reply:
Done!